src/libraries/Common/tests/System/IO/Compression/CompressionStreamUnitTestBase.cs

```diff
@@ -479,6 +478,8 @@ async Task<long> GetLengthAsync(CompressionLevel compressionLevel)
         [InlineData(TestScenario.ReadByteAsync)]
         public async Task StreamTruncation_IsDetected(TestScenario scenario)
         {
+            AppContext.SetSwitch("System.IO.Compression.UseStrictValidation", true);
```

Comment:
This affects all tests process-wide. Any test that changes such a process-wide switch needs to use RemoteExecutor to move that code out of process.

Reply:
Ok, done.

Reply:
Another option is to create a second test project that enables the appcontext switch and runs all tests of the "normal" project with the switch on.

Reply:
And a third option could be to turn the switch on for all the tests, we already know our tests will pass with the "strict validation" in place.

Reply:
> a third option could be to turn the switch on for all the tests, we already know our tests will pass with the "strict validation" in place.

Not anymore, this one will fail https://github.com/dotnet/runtime/blob/eead714e4a5e82f2075de7d5eaad75dedf509941/src/libraries/System.IO.Compression/tests/CompressionStreamUnitTests.Deflate.cs#L78

> Another option is to create a second test project that enables the appcontext switch and runs all tests of the "normal" project with the switch on.

I prefer the current way (with the RemoteExecutor) but no strong feelings. Let me know.